### PR TITLE
Issue 628: GuiFont! fails for bad monospace fonts

### PR DESCRIPTION
**Pull request by jgehrig**
Issue #628

The line following line from setAttaced is invalid:
	`m_nvim->api0()->vim_set_option("guifont", fontDesc());`

This works for well formed monospace fonts, but does not work for bad
monospace fonts. This line updates the value of `guifont` with the default
font (valid monospace) on initialization.

If the user then sets a bad monospace font with GuiFont!, in some cases timing
of the neovim guifont events can override the user defined font.

### CODE REVIEW

**Comment by Mte90** [comment]
Confirmed that works :-)

File: src/gui/shell.cpp [hunk 1]
```diff
@@ -852,14 +850,21 @@ void Shell::handleMouse(bool enabled)
...
 {
-	if (args.size() == 2) {
-		QString fdesc = m_nvim->decode(args.at(1).toByteArray());
-		setGuiFont(fdesc, false /*force*/, true /*setOption*/);
-	} else if (args.size() == 3) {
-		QString fdesc = m_nvim->decode(args.at(1).toByteArray());
-		const bool force = args.at(2) == 1;
-		setGuiFont(fdesc, force, true /*setOption*/);
+	if (args.size() < 2
+		|| !args.at(0).canConvert<QByteArray>())
```
Comment (jgehrig) [inline src/gui/shell.cpp]: This should be `!args.at(1).canConvert<QByteArray>()`

**Comment by jgehrig** [comment]
@equalsraf 

Ready for merge. Let's get this checked-in to unblock the issue.

**Comment by equalsraf** [comment]
Merged, thanks @jgehrig

**Pull request merged**